src/commonMain/kotlin/io/ktor/io/ByteArrayBufferPool.kt

```diff
@@ -4,21 +4,17 @@ public class ByteArrayBufferPool(
     capacity: Int = DEFAULT_POOL_CAPACITY,
 ) : DefaultPool<ByteArrayBuffer>(capacity) {
 
-    override fun produceInstance(): ByteArrayBuffer = ByteArrayBuffer(DEFAULT_BUFFER_SIZE)
+    override fun produceInstance(): ByteArrayBuffer = ByteArrayBuffer(ByteArrayPool.Default)
 
     override fun clearInstance(instance: ByteArrayBuffer): ByteArrayBuffer {
-        instance.reset()
+        instance.resetForWrite()
         return instance
     }
 
     public companion object {
         public val Default: ObjectPool<ByteArrayBuffer> = ByteArrayBufferPool()
 
-        public val NoPool: NoPoolImpl<ByteArrayBuffer> = object : NoPoolImpl<ByteArrayBuffer>() {
-            override fun borrow(): ByteArrayBuffer {
-                throw NotImplementedError()
-            }
-        }
+        public val Empty: ObjectPool<ByteArrayBuffer> = ByteArrayBufferPool(capacity = 0)
```

Comment:
is it expected, that even if this pool is empty, after closing buffer borrowed from empty pool, it will be returned to ByteArrayPool.Default?
And also almost all other `Empty` pools will work in the same way

Reply:
Fixed